lib/libvcc/vcc_xref.c

```diff
@@ -326,6 +326,29 @@ vcc_CheckUses(struct vcc *tl)
 
 /*---------------------------------------------------------------------*/
 
+static void v_matchproto_(symwalk_f)
+vcc_instance_info(struct vcc *tl, const struct symbol *sym)
+{
+
+	CHECK_OBJ_NOTNULL(sym, SYMBOL_MAGIC);
+	CHECK_OBJ_NOTNULL(sym->kind, KIND_MAGIC);
+	CHECK_OBJ_NOTNULL(sym->type, TYPE_MAGIC);
+	if (sym->rname == NULL || sym->kind != SYM_INSTANCE)
```

Comment:
Can a `SYM_INSTANCE` not have an `rname`?

Reply:
I guess you might have misread the code. We do only add to the instance info `SYM_INSTANCE` _with_ an `rname`.

Reply:
Let me rephrase, in which case would a `SYM_INSTANCE` not have an `rname`?

Reply:
I meant that my understanding of the symbol table for this case is this:

```diff
--- lib/libvcc/vcc_xref.c
+++ lib/libvcc/vcc_xref.c
@@ -333,8 +333,9 @@ vcc_instance_info(struct vcc *tl, const struct symbol *sym)
        CHECK_OBJ_NOTNULL(sym, SYMBOL_MAGIC);
        CHECK_OBJ_NOTNULL(sym->kind, KIND_MAGIC);
        CHECK_OBJ_NOTNULL(sym->type, TYPE_MAGIC);
-       if (sym->rname == NULL || sym->kind != SYM_INSTANCE)
+       if (sym->kind != SYM_INSTANCE)
                return;
+       AN(sym->rname);
        Fc(tl, 1, "\t{ .p = &%s, .name = \"", sym->rname);
        VCC_SymName(tl->fc, sym);
        Fc(tl, 1, "\" },\n");
```

Reply:
Do we want such assertions for code which only generates linkage for debugging?

Reply:
anyway, nevermind. I will just add your suggestion, it's not worth it to discuss

Reply:
Also to be consistent we shouldn't auto-indent the generated code:

```diff
--- lib/libvcc/vcc_xref.c
+++ lib/libvcc/vcc_xref.c
@@ -332,12 +332,12 @@ vcc_instance_info(struct vcc *tl, const struct symbol *sym)
 
        CHECK_OBJ_NOTNULL(sym, SYMBOL_MAGIC);
        CHECK_OBJ_NOTNULL(sym->kind, KIND_MAGIC);
-       CHECK_OBJ_NOTNULL(sym->type, TYPE_MAGIC);
-       if (sym->rname == NULL || sym->kind != SYM_INSTANCE)
+       if (sym->kind != SYM_INSTANCE)
                return;
+       AN(sym->rname);
-       Fc(tl, 1, "\t{ .p = &%s, .name = \"", sym->rname);
+       Fc(tl, 0, "\t{ .p = &%s, .name = \"", sym->rname);
        VCC_SymName(tl->fc, sym);
-       Fc(tl, 1, "\" },\n");
+       Fc(tl, 0, "\" },\n");
 }
 
 void
```

This doesn't change the output, but it's consistent with how the end-of-ii marker is generated.